pkg/storage/bloom/v1/index.go

```diff
@@ -361,25 +280,29 @@ func (d *SeriesPageDecoder) Err() error {
 	return d.dec.Err()
 }
 
+// series encoding/decoding --------------------------------------------------
+
 type Series struct {
 	Fingerprint model.Fingerprint
 	Chunks      ChunkRefs
 }
 
-// SeriesWithOffsets is a series with a a variable number
-// of bloom offsets. Used in v2+ to store blooms for larger series
-// in parts
-type SeriesWithOffsets struct {
+type Meta struct {
+	Fields  Set[Field]
 	Offsets []BloomOffset
+}
+
+type SeriesWithMeta struct {
 	Series
+	Meta
 }
 
-func (s *SeriesWithOffsets) Encode(
+func (s *SeriesWithMeta) Encode(
 	enc *encoding.Encbuf,
+	version Version,
```

Comment:
What do you expect we'll end up using the version for? Right now nothing on this call path actually uses it, right?

Reply:
The reason was to keep the `Encode()` and `Decode` functions for the different types consistent:

```
Encode(enc *encoding.Encbuf, v Version, ... other arguments)
Decode(dec *encoding.Decbuf, v Version, ... other arguments)
```

Reply:
E.g the version is now passed to `offset.Encode` on line 314